jmx-metrics/src/main/resources/target-systems/solr.groovy

```diff
@@ -0,0 +1,81 @@
+  "Value", otel.&longUpDownCounterCallback)
+
+def beanSolrCoreRequests = otel.mbeans(["solr:dom1=core,dom2=*,category=QUERY,scope=*,name=requests",
+                                        "solr:dom1=core,dom2=*,category=UPDATE,scope=*,name=requests"])
+otel.instrument(beanSolrCoreRequests, "solr.request.count", "The number of queries made.", "queries",
+  ["core" : { mbean -> mbean.name().getKeyProperty("dom2") },
+   "type" : { mbean -> mbean.name().getKeyProperty("category") },
+   "handler" : { mbean -> mbean.name().getKeyProperty("scope") }],
+  "Count", otel.&longCounterCallback)
+
+def beanSolrCoreRequestTimes = otel.mbeans(["solr:dom1=core,dom2=*,category=QUERY,scope=*,name=requestTimes",
+                                        "solr:dom1=core,dom2=*,category=UPDATE,scope=*,name=requestTimes"])
+otel.instrument(beanSolrCoreRequestTimes, "solr.request.time.average",
+  "The average time of a query. This average is based on the histogram configuration in Solr. By default, Solr uses an exponentially decaying reservoir.",
+  "ms",
+  ["core" : { mbean -> mbean.name().getKeyProperty("dom2") },
+   "type" : { mbean -> mbean.name().getKeyProperty("category") },
+   "handler" : { mbean -> mbean.name().getKeyProperty("scope") }],
+  "Mean", otel.&doubleValueCallback)
+
+def beanSolrCoreErrors = otel.mbeans(["solr:dom1=core,dom2=*,category=QUERY,scope=*,name=errors",
+                                            "solr:dom1=core,dom2=*,category=UPDATE,scope=*,name=errors"])
+otel.instrument(beanSolrCoreErrors, "solr.request.error.count", "The number of queries resulting in an error.", "queries",
+  ["core" : { mbean -> mbean.name().getKeyProperty("dom2") },
+   "type" : { mbean -> mbean.name().getKeyProperty("category") },
+   "handler" : { mbean -> mbean.name().getKeyProperty("scope") }],
+  "Count", otel.&longCounterCallback)
+
+def beanSolrCoreTimeouts = otel.mbeans(["solr:dom1=core,dom2=*,category=QUERY,scope=*,name=timeouts",
+                                      "solr:dom1=core,dom2=*,category=UPDATE,scope=*,name=timeouts"])
+otel.instrument(beanSolrCoreTimeouts, "solr.request.timeout.count", "The number of queries resulting in a timeout.", "queries",
+  ["core" : { mbean -> mbean.name().getKeyProperty("dom2") },
+   "type" : { mbean -> mbean.name().getKeyProperty("category") },
+   "handler" : { mbean -> mbean.name().getKeyProperty("scope") }],
+  "Count", otel.&longCounterCallback)
+
+def beanSolrCoreQueryResultsCache = otel.mbean("solr:dom1=core,dom2=*,category=CACHE,scope=*,name=queryResultCache")
+otel.instrument(beanSolrCoreQueryResultsCache, "solr.cache.eviction.count", "The number of evictions from a cache.", "evictions",
+  ["core" : { mbean -> mbean.name().getKeyProperty("dom2") },
+   "cache" : { mbean -> mbean.name().getKeyProperty("scope") }],
+  "cumulative_evictions", otel.&longCounterCallback)
+otel.instrument(beanSolrCoreQueryResultsCache, "solr.cache.hit.count", "The number of hits for a cache.", "hits",
```

Comment:
Feels like hits/inserts could be combined to one metric of "solr.cache.lookups.count" and some kind of label of "hit/miss", and lookups would be unnecessary.

Reply:
@dehaansa miss actually doesn't appear to be a metric that we can directly get, so i think it's ok as is. i see your reasoning though

Reply:
Sounds good.